### PR TITLE
[HELM] Fix Ingress Resource of Meshery

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

This PR:
1. Fixes the Ingress Resource Template for Meshery
2. Adds missing support for IngressClassName
3. Improves dynamic lookup for Ingress APIVersion

<details>
<summary>Dry Run Response</summary>
<br>

```
NAME: meshery
LAST DEPLOYED: Thu Dec 28 19:53:18 2023
NAMESPACE: meshery
STATUS: pending-install
REVISION: 1
TEST SUITE: None
HOOKS:
MANIFEST:
---
# Source: meshery/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: meshery-server
  namespace: meshery
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
---
# Source: meshery/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: meshery-server
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
- nonResourceURLs: ["/metrics", "/health", "/ping"]
  verbs:
  - get
---
# Source: meshery/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: meshery-server
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
roleRef:
  kind: ClusterRole
  name: meshery-server
  apiGroup: rbac.authorization.k8s.io
subjects:
- kind: ServiceAccount
  name: meshery-server
  namespace: meshery
---
# Source: meshery/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: meshery
  namespace: meshery
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
  annotations:
    {}
spec:
  type: LoadBalancer
  ports:
    - port: 9081
      targetPort: 8080
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
---
# Source: meshery/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: meshery
  namespace: meshery
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
  annotations:
    {}
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: meshery
      app.kubernetes.io/instance: meshery
  template:
    metadata:
      labels:
        app.kubernetes.io/name: meshery
        app.kubernetes.io/instance: meshery
    spec:
      restartPolicy: Always
      serviceAccountName: meshery-server
      securityContext:
        {}
      containers:
        - name: meshery
          securityContext:
            {}
          image: "layer5/meshery:stable-latest"
          imagePullPolicy: Always
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
          env:
          - name: ADAPTER_URLS
            value: meshery-istio:10000 meshery-linkerd:10001 meshery-consul:10002 meshery-kuma:10007 meshery-nginx-sm:10010 meshery-nsm:10004 meshery-app-mesh:10005 meshery-traefik-mesh:10006 meshery-cilium:10012
          - name: EVENT
            value: mesheryLocal
          - name: MESHERY_SERVER_CALLBACK_URL
            value: 
          - name: PROVIDER
            value: 
          - name: PROVIDER_BASE_URLS
            value: https://meshery.layer5.io
          resources:
            {}
---
# Source: meshery/templates/ingress.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: meshery
  namespace: meshery
  labels:
    helm.sh/chart: meshery-0.7.0
    app.kubernetes.io/name: meshery
    app.kubernetes.io/instance: meshery
    app.kubernetes.io/managed-by: Helm
spec:
  rules:
    - host: "meshery.local.com"
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: meshery
                port:
                  number: 9081

NOTES:
1. Get the application URL by running these commands:
  http://meshery.local.commap[path:/ pathType:Prefix]
```

</details>

Screenshot 2023-12-28 at 7 54 25 PM

Screenshot 2023-12-28 at 7 55 29 PM

Screenshot 2023-12-28 at 7 56 12 PM

Screenshot 2023-12-28 at 8 11 26 PM

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: install/kubernetes/helm/meshery/templates/ingress.yaml [hunk 1]
```diff
@@ -29,16 +35,25 @@ spec:
...
+    {{- range .Values.ingress.hosts }}
+    - host: {{ tpl .host $ | quote }}
       http:
         paths:
-        {{- range .paths }}
-          - path: {{ . }}
-            pathType: {{ $pathType }}
+          {{- range .paths }}
+          - path: {{ .path }}
+            {{- if and .pathType (eq $apiVersion "networking.k8s.io/v1") }}
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/templates/ingress.yaml:44]: Is this comparing pathType and apiVersion?

Comment (leecalcote) [reply install/kubernetes/helm/meshery/templates/ingress.yaml:44]: If so, pathType will either be "ImplementationSpecific" or "Prefix"

Comment (theBeginner86) [reply install/kubernetes/helm/meshery/templates/ingress.yaml:44]: No, its checking if the `apiVersion` is "networking.k8s.io/v1" AND `.pathType` is specified.

Comment (theBeginner86) [reply install/kubernetes/helm/meshery/templates/ingress.yaml:44]: I have already added a default pathType as Prefix in values.yaml 👍

File: install/kubernetes/helm/meshery/values.yaml [hunk 2]
```diff
@@ -60,25 +60,29 @@ service:
...
     # kubernetes.io/tls-acme: "true"
 
+  ## apiVersion:
+
   hosts:
-    - host: chart-example.local
-      
-      ## Paths to use for ingress rules
-      paths: []
+    - host: meshery.local.com
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/values.yaml:73]:
```suggestion
    - host: meshery.local
```

**Review by leecalcote — APPROVED** [review]

**Pull request merged**